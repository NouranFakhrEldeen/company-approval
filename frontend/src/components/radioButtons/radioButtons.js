import React from 'react';
import './radioButtons.scss';

export function RadioButtons({
  name,
  value,
  options,
  className,
  onChange,
  disabled,
  id,
}) {
  return (
    <div id={id} className={`${disabled ? 'disabled' : ''} radio-button-component ${className}`}>
      {options?.map((option, index)=>(
        <label
          className={`${option.value === value ? 'radio-selected' : 'radio-not-selected'} radio-button-item`}
          key={`${option.value}_${index}`}>
          <input
            disabled={disabled}
            type="radio"
            value={option.value}
            name={name}
            checked={option.value === value}
            onChange={onChange}
          />
          &nbsp;
          <span>{option.label}</span>
        </label>
      ))}
    </div>
  );
}
