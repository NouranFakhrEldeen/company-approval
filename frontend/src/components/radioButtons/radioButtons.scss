@import '../../styles/variables.scss';

.radio-button-component {
  display: inline-block;
  position: relative;
  width: 100%;
  label {
    position: relative;
    color: #fff;
    padding: 8px 8px;
    min-width: 100px;
    width: 100%;
    border: 1px solid #ccc;
  }
  .radio-default-select {
    background-color: #285cb9;
    color: #fff;
  }
  .radio-selected {
    background-color: #285cb9;
    color: #fff;
  }
  .radio-not-selected {
    background-color: #f5f6f8;
    color: #000;
  }
  label span {
    width: calc(100% - 30px);
    white-space: nowrap; 
    overflow: hidden;
    text-overflow: ellipsis; 
    color: unset;
    font-weight: normal;
    font-size: 0.9em;
    display: inline-block;
  }
  .vertical-radio-buttons{
    position: relative;
    width: 100%;
  }
}
.radio-button-component.disabled{
  .radio-not-selected {
    background-color: transparent;
    color: #000;
  }
}
