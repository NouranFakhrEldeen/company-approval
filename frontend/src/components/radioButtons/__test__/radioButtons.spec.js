import React from 'react';
import { render, fireEvent, cleanup } from '@testing-library/react';
// import { HashRouter as Router } from 'react-router-dom';
import renderer from 'react-test-renderer';
import { RadioButtons } from '../radioButtons';

let onChange = jest.fn();
let radioBtnArray = [
  { label: 'yes', value: 'YES', name: 'first' },
  { label: 'no', value: 'NO', name: 'second' },
];
describe('testing delayed input', ()=>{
  afterEach(cleanup);


  //   test('should checkbox checked', () => {
  //     const { getByTestId } = render(<DelayedInput data-testid='delayed-input-it' onChange={()=> onChange()}/>);
  //     const input = getByTestId('delayed-input-it');
  //     fireEvent.change(input, { target: { value: '23' } });
  //     expect(input.value).toBe('23');
  //   });

  test('should test radio buttons', () => {
    const { getByLabelText } = render(
      <form>
        <label>
             First <input type="radio" name="radio1" value="first" />
        </label>
        <label>
            Second <input type="radio" name="radio1" value="second" />
        </label>
      </form>
    );

    const radio = getByLabelText('First');
    fireEvent.change(radio, { target: { value: 'second' } });
    expect(radio.value).toBe('second');
  });


  //   snapshot
  test('should do snapshot', () => {
    const radiobutton = renderer.create(
      <RadioButtons
        disabled={''}
        className='col-12 padding-0'
        name={'answer'}
        value={''}
        options={radioBtnArray}
        onChange={()=> onChange()}
      />
    ).toJSON();
    expect(radiobutton).toMatchSnapshot();
  });

});