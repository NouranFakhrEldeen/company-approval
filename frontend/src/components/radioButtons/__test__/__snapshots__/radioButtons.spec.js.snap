// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing delayed input should do snapshot 1`] = `
<div
  className=" radio-button-component col-12 padding-0"
>
  <label
    className="radio-not-selected radio-button-item"
  >
    <input
      checked={false}
      disabled=""
      name="answer"
      onChange={[Function]}
      type="radio"
      value="YES"
    />
     
    <span>
      yes
    </span>
  </label>
  <label
    className="radio-not-selected radio-button-item"
  >
    <input
      checked={false}
      disabled=""
      name="answer"
      onChange={[Function]}
      type="radio"
      value="NO"
    />
     
    <span>
      no
    </span>
  </label>
</div>
`;
